### PR TITLE
Fix failure on blank image URLs passed from oauth

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

All our auth services have shown to _occasionally_ fails authentication because they can pass a profile which has no remote image string.

This creates a `Users::SafeRemoteProfileImageUrl` which accepts a URL if passed, but falls back to our default generator if not.

It does not attempt to fully download and validate the image which I think would be redundant with the duties of carrierwave, and is plenty to fix this particular case without doing the extra work of downloading the image twice.

## Related Tickets & Documents

Closes https://github.com/forem/internalEngineering/issues/278

## QA Instructions, Screenshots, Recordings

A manual QA approach is described in the issue.

### UI accessibility concerns?

No

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### CODE REVIEW

File: app/services/users/safe_remote_profile_image_url.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module Users
+  module SafeRemoteProfileImageUrl
+    # Basic check for nil and blank URLs, alongside likely incomplete URLs, such as just "image.jpg".
+    def self.call(url)
+      return Users::ProfileImageGenerator.call unless url.to_s.start_with?("https")
```
Comment (citizen428) [inline app/services/users/safe_remote_profile_image_url.rb:5]: Maybe it's just me, but I would find this easier to read if you reverse the logic:

```ruby
return url if url.to_s.start_with?("https")

Users::ProfileImageGenerator.call
```

Comment (benhalpern) [reply app/services/users/safe_remote_profile_image_url.rb]: Changed!

**Comment by Zhao-Andy** [comment]
Possibly resolves #12440

**Review by Zhao-Andy — APPROVED** [review]
LGTM!

**Review by citizen428 — APPROVED** [review]

**Pull request merged**